### PR TITLE
Add support for generating test vectors for entity metadata descriptors

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

**Review by kostko — APPROVED** [review]

File: testvectors/testvectors.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	return EntityMetadataTestVector{
+		Kind:                    kind,
+		SignatureContext:        string(sigCtx),
+		EntityMeta:              *meta,
+		SignedEntityMeta:        *sigMeta,
+		EncodedEntityMeta:       cbor.Marshal(meta),
+		EncodedSignedEntityMeta: cbor.Marshal(sigMeta),
+		Valid:                   valid,
```
Comment (ptrus) [inline testvectors/testvectors.go:56]: Doesn't the `Valid` field in the `*TestVector` struct refer to "is this a valid signature", which i believe should be `True` even for signatures of "invalid metadata registry content" (which i think is kinda irrelevant for the test-vectors as that's just a registry limit, and the signer should probably handle correctly both).

Comment (tjanez) [reply testvectors/testvectors.go:56]: I discussed this with @kostko and the `valid` field for the [Oasis Core's Transaction Test Vectors](https://docs.oasis.dev/oasis-core/high-level-components/index/test-vectors) is defined quite broadly :slightly_smiling_face: :

> `valid` is a boolean flag indicating whether the given test vector represents a valid transaction.

That being said, I agree that checking something like the validity of an URL, email, Keybase or Twitter scheme would be an overkill for a HSM signer app.

I'll split the test cases into 2, ones that are simple size checks and the other ones that check the semantics of a field's value, and update `gen_vectors` to only create test vectors with the first set.

Comment (tjanez) [reply testvectors/testvectors.go:56]: @kostko, @ptrus, I've updated the PR, plese take another look.

Comment (ptrus) [reply testvectors/testvectors.go:56]: Yeah I think this makes more sense now (test vectors not including all of the semantic checks). Thanks.

Comment (tjanez) [reply testvectors/testvectors.go:56]: > Yeah I think this makes more sense now (test vectors not including all of the semantic checks). Thanks.

No problem. Thanks for noticing and flagging this yesterday!

**Review by kostko — APPROVED** [review]

**Pull request merged**